sw/applications/example_spi_obi_slave/main.c

```diff
@@ -0,0 +1,208 @@
+    // The RX Queue is empty, thus could not read from RX register
+    SPI_SLAVE_FLAG_RX_QUEUE_EMPTY     = 0x0040,
+    // The SPI is not ready
+    SPI_SLAVE_FLAG_NOT_READY          = 0x0080,
+    // The event to enable is not a valid event
+    SPI_SLAVE_FLAG_EVENT_INVALID      = 0x0100,
+    // The error irq to enable is not a valid error irq
+    SPI_SLAVE_FLAG_ERROR_INVALID      = 0x0200 
+} spi_slave_flags_e;
+
+spi_host_t* spi; //Removed specific memory assignement
+
+
+spi_slave_flags_e spi_slave_init(spi_host_t* spi_host);
+void spi_host_write();
+spi_slave_flags_e spi_slave_read(uint32_t addr, void* data, uint16_t length);
+bool check_address_validity(uint32_t addr, void* data, uint16_t length);
+
+
+int main(int argc, char *argv[])
+{   
+    spi_return_flags_e flags;
+    flags = spi_slave_init(spi);
+    if (flags != SPI_FLAG_OK){
+        printf("Failure to initialize\n Error code: %s", flags);
+        return EXIT_FAILURE;
+    }
+
+
+
+
+    return EXIT_SUCCESS;
+}
+
+void spi_host_write(){
+};
+
+spi_slave_flags_e spi_slave_read(uint32_t addr, void* data, uint16_t length){
+    // ??? Is this necessary? What happens when address is invalid or data exceeds memory? What is the size of the memory?
+    if (check_address_validity(addr, data, length) != true) return SPI_SLAVE_FLAG_ADDRESS_INVALID;
+
+
+    uint32_t wrap_length_cmd    = (WRITE_SPI_SLAVE_REG_2)                   //Write register 2 
+                                + ((LOWER_BYTE_16_BITS(length)) << 8)       //Wraplength low 
+                                + (WRITE_SPI_SLAVE_REG_3 << 16)             //Write register 3
+                                + (UPPER_BYTE_16_BITS(length) << 24);       //Wraplength high
+
+    // Load command to TX FIFO
+    spi_write_word(spi, wrap_length_cmd);
+    spi_wait_for_ready(spi);
+
+    // Set up segment parameters -> send commands
+    const uint32_t send_cmd_byte = spi_create_command((spi_command_t){
+        .len        = 1,                    // 1 Byte (The SPI SLAVE IP can only take one CMD byte at a time)
+        .csaat      = true,                 // Command not finished e.g. CS remains low after transaction
+        .speed      = SPI_SPEED_STANDARD,   // Single speed
+        .direction  = SPI_DIR_TX_ONLY       // Write only
+    });
+
+    for(int i = 0; i < 4; i++){ //To test but it might be possible to delete this and put .len in send_cmd_byte to 4. But not sure if that works as intended
+        // Load segment parameters to COMMAND register
+        spi_set_command(spi, send_cmd_byte);
+        spi_wait_for_ready(spi);
+    }
+
+    spi_write_byte(spi, READ_SPI_SLAVE_CMD);
+    spi_wait_for_ready(spi);
+
+    spi_set_command(spi, send_cmd_byte);
+    spi_wait_for_ready(spi);
+
+    //write address
+    spi_write_word(spi, addr);
+    spi_wait_for_ready(spi);
+
+    const uint32_t send_cmd_word = spi_create_command((spi_command_t){
+      .len        = 4,                     // 4 Bytes 
```

Comment:
I think that if you want to write N bytes you have to pass to the function N-1, look at the examples